org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/ProjectDescription.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2015, 2017 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.resource.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+import com.google.common.annotations.Beta;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * 
+ * @since 2.9
+ */
+@Beta
+public class ProjectDescription {
+
+	/**
+	 * A unique name for this project
+	 */
+	private String name;
+
+	/**
+	 * list of logical names of upstream dependencies
+	 */
+	private List<String> dependencies = new ArrayList<>();
+
+	public static ProjectDescription findInEmfObject(Notifier emfObject) {
+		for (Adapter adapter : emfObject.eAdapters()) {
+			if (adapter instanceof ProjectDescriptionAdapter)
+				return ((ProjectDescriptionAdapter) adapter).get();
+		}
+		return null;
+	}
+
+	public static ProjectDescription removeFromEmfObject(Notifier emfObject) {
+		List<Adapter> adapters = emfObject.eAdapters();
+		for (int i = 0, max = adapters.size(); i < max; i++) {
+			Adapter adapter = adapters.get(i);
+			if (adapter instanceof ProjectDescriptionAdapter) {
+				emfObject.eAdapters().remove(i);
```

Comment:
Iterator please

Reply:
Please also update `@EmfAdaptable`

Reply:
No, please don't.
Catfight :)

Reply:
I have removed the max but stay with ordinary for loop. Hope all can live with that (*MIAOU*).